beamer/tests/contracts/test_request_manager.py

```diff
@@ -99,6 +99,85 @@ def test_claim_with_different_stakes(token, request_manager, claim_stake):
         request_manager.claimRequest(request_id, FILL_ID, {"from": claimer})
 
 
+def test_claim_on_behalf_of_other(token, request_manager, claim_stake, claim_period):
+    """
+    Test that making a claim on behalf of others creates correct claim
+    and claimer can withdraw afterwards
+    """
+    (
+        requester,
+        initiator,
+    ) = alloc_accounts(2)
+    (claimer,) = alloc_whitelisted_accounts(1, {request_manager})
+
+    transfer_amount = 92
+    initiator_eth_balance = web3.eth.get_balance(initiator.address)
+    claimer_eth_balance = web3.eth.get_balance(claimer.address)
+
+    token.mint(requester, transfer_amount, {"from": requester})
+    assert token.balanceOf(requester) == transfer_amount
+    assert token.balanceOf(initiator) == 0
+    assert token.balanceOf(claimer) == 0
+
+    request_id = make_request(
+        request_manager,
+        token=token,
+        requester=requester,
+        target_address=requester,
+        amount=transfer_amount,
+    )
+
+    claim_tx = request_manager.claimRequest(
+        claimer, request_id, FILL_ID, {"from": initiator, "value": claim_stake}
+    )
+    claim_id = claim_tx.return_value
+    expected_termination = (
+        request_manager.claimPeriod() + web3.eth.get_block("latest")["timestamp"]
+    )
+
+    assert "ClaimMade" in claim_tx.events
+    claim_event = claim_tx.events["ClaimMade"]
+    assert RequestId(claim_event["requestId"]) == request_id
+    assert claim_event["claimId"] == claim_id
+    assert claim_event["claimer"] == claimer
+    assert claim_event["claimerStake"] == claim_stake
+    assert claim_event["lastChallenger"] == brownie.ZERO_ADDRESS
+    assert claim_event["challengerStakeTotal"] == 0
+    assert claim_event["termination"] == expected_termination
+    assert claim_event["fillId"] == to_hex(FILL_ID)
+
+    assert web3.eth.get_balance(request_manager.address) == claim_stake
+    assert web3.eth.get_balance(initiator.address) == initiator_eth_balance - claim_stake
+    assert web3.eth.get_balance(claimer.address) == claimer_eth_balance
+
+    chain.mine(timedelta=claim_period)
```

Comment:
it would be nice if chain.sleep is used to time travel

Reply:
We use chain.mine basically everywhere in the test to time travel. Should it be replaced everywhere?

Reply:
ideally yes. But I think chain.sleep is limited to timedelta. So don't spend too much time on it

Reply:
We have #1229, so from my side this will be tackled anyway. You don't need to change here anything, especially with #1079 in the making